component/grant-type/src/main/java/org/wso2/carbon/identity/oauth2/grant/token/exchange/TokenExchangeGrantHandler.java

```diff
@@ -0,0 +1,326 @@
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.oauth.common.OAuth2ErrorCodes;
+import org.wso2.carbon.identity.oauth.config.OAuthServerConfiguration;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.dto.OAuth2AccessTokenRespDTO;
+import org.wso2.carbon.identity.oauth2.grant.token.exchange.utils.TokenExchangeUtils;
+import org.wso2.carbon.identity.oauth2.model.RequestParameter;
+import org.wso2.carbon.identity.oauth2.token.OAuthTokenReqMessageContext;
+import org.wso2.carbon.identity.oauth2.token.handlers.grant.AbstractAuthorizationGrantHandler;
+import org.wso2.carbon.identity.oauth2.util.ClaimsUtil;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.wso2.carbon.identity.oauth2.grant.token.exchange.utils.TokenExchangeUtils.handleException;
+
+/**
+ * Class to handle Token Exchange grant type.
+ */
+public class TokenExchangeGrantHandler extends AbstractAuthorizationGrantHandler {
+
+    private static final Log log = LogFactory.getLog(TokenExchangeGrantHandler.class);
+    private int validityPeriod;
+    private boolean validateIAT = true;
+    private String[] registeredClaimNames = new String[]{"iss", "sub", "aud", "exp", "nbf", "iat", "jti"};
+    private String requested_token_type = TokenExchangeConstants.JWT_TOKEN_TYPE;
+
+    /**
+     * Initialize the TokenExchangeGrantHandler
+     *
+     * @throws IdentityOAuth2Exception Error when initializing
+     */
+    public void init() throws IdentityOAuth2Exception {
+
+        super.init();
+
+        String validateIATProp = IdentityUtil.getProperty(TokenExchangeConstants.PROP_ENABLE_IAT_VALIDATION);
+        validateIAT = Boolean.parseBoolean(validateIATProp);
+
+        String validityPeriodProp = IdentityUtil.getProperty(TokenExchangeConstants.PROP_IAT_VALIDITY_PERIOD);
+
+        if (validateIAT) {
+            if (StringUtils.isNotBlank(validityPeriodProp)) {
+                try {
+                    validityPeriod = Integer.parseInt(validityPeriodProp);
+                } catch (NumberFormatException e) {
+                    validityPeriod = TokenExchangeConstants.DEFAULT_IAT_VALIDITY_PERIOD;
+                    log.warn("Invalid value: " + validityPeriodProp + " is set for IAT validity period. Using " +
+                            "default value: " + validityPeriod + " minutes.");
+                }
+            } else {
+                validityPeriod = TokenExchangeConstants.DEFAULT_IAT_VALIDITY_PERIOD;
+                log.warn("Empty value is set for IAT validity period. Using default value: " + validityPeriod
+                        + " minutes.");
+            }
```

Comment:
To improve readability of the init method, how about moving this to a private method that return the validityPeriod